extensions/wikia/Recirculation/RecirculationHooks.class.php

```diff
@@ -100,51 +102,73 @@ static public function canShowDiscussions( $cityId, $ignoreWgEnableRecirculation
-		       $wgStagingEnvironment,
-		       $wgWikiaEnvironment;
+		$metaData = self::getMetaData();
+		$metaDataJson = json_encode( $metaData );
+
+		$outputPage->addScript(
+			"<script id=\"liftigniter-metadata\" type=\"application/json\">${metaDataJson}</script>"
+		);
+	}
 
+	private static function getMetaData() {
+		global $wgLanguageCode, $wgCityId, $wgEnableArticleFeaturedVideo;
 		$context = RequestContext::getMain();
 		$title = $context->getTitle();
 		$articleId = $title->getArticleID();
-		$metaData = [];
+		$shouldNoIndex = self::shouldIndex();
 		$metaDataService = new LiftigniterMetadataService();
 		$metaDataFromService = $metaDataService->getLiMetadataForArticle( $wgCityId, $articleId );
-
 		$metaData['language'] = $wgLanguageCode;
 
 		if ( !empty( $metaDataFromService ) ) {
 			$metaData['guaranteed_impression'] = $metaDataFromService->getGuaranteedNumber();
-			$metaData['start_date'] = $metaDataFromService->getDateFrom()->format('Y-m-d H:i:s');
-			$metaData['end_date'] = $metaDataFromService->getDateTo()->format('Y-m-d H:i:s');
+			$metaData['start_date'] = $metaDataFromService->getDateFrom()->format( 'Y-m-d H:i:s' );
+			$metaData['end_date'] = $metaDataFromService->getDateTo()->format( 'Y-m-d H:i:s' );
 			if ( !empty( $metaDataFromService->getGeos() ) ) {
 				$metaData['geolocation'] = $metaDataFromService->getGeos();
 			}
 		}
 
-		$isProduction =
-			empty( $wgDevelEnvironment ) && empty( $wgStagingEnvironment ) &&
-			$wgWikiaEnvironment !== WIKIA_ENV_STAGING;
-		$isPrivateWiki = WikiFactory::isWikiPrivate( $wgCityId ) || $wgIsPrivateWiki;
-
-		if ( !$isProduction || $isPrivateWiki || $title->inNamespace( NS_FILE ) ) {
+		if ( $shouldNoIndex ) {
 			$metaData['noIndex'] = 'true';
 		}
 
 		if ( !empty( $wgEnableArticleFeaturedVideo ) &&
-		     ArticleVideoContext::isFeaturedVideoEmbedded( $title->getPrefixedDBkey() )
+			ArticleVideoContext::isFeaturedVideoEmbedded( $title->getPrefixedDBkey() )
 		) {
 			$metaData['type'] = 'video';
 		}
 
-		$metaDataJson = json_encode( $metaData );
+		return $metaData;
+	}
 
-		$outputPage->addScript(
-			"<script id=\"liftigniter-metadata\" type=\"application/json\">${metaDataJson}</script>"
-		);
+	private static function checkIfIsProduction() {
+		global $wgDevelEnvironment, $wgStagingEnvironment, $wgWikiaEnvironment;
+
+		return empty( $wgDevelEnvironment ) &&
+			empty( $wgStagingEnvironment ) &&
+			$wgWikiaEnvironment !== WIKIA_ENV_STAGING;
+	}
+
+	private static function shouldIndex() {
+		return self::isPrivateOrProduction() || self::isCorrectNameSpace();
+	}
+
+	private static function isPrivateOrProduction() {
+		global $wgCityId, $wgIsPrivateWiki;
+
+		$isProduction = self::checkIfIsProduction();
+		$isPrivateWiki = WikiFactory::isWikiPrivate( $wgCityId ) || $wgIsPrivateWiki;
+
+		return !$isProduction || $isPrivateWiki;
+	}
+
+	private static function isCorrectNameSpace() {
+		$context = RequestContext::getMain();
+		$title = $context->getTitle();
```

Comment:
imo there's no need to `$context` var here and `$title = RequestContext::getMain()->getTitle()` would be fine